x-pack/legacy/plugins/alerting/server/types.ts

```diff
@@ -76,13 +77,17 @@ export interface Alert {
   scheduledTaskId?: string;
   createdBy: string | null;
   updatedBy: string | null;
+  createdAt: Date;
+  updatedAt: Date | null;
   apiKey: string | null;
   apiKeyOwner: string | null;
   throttle: string | null;
   muteAll: boolean;
   mutedInstanceIds: string[];
 }
 
+export type PartialAlert = Pick<Alert, 'id'> & Partial<Omit<Alert, 'id'>>;
```

Comment:
could this be shorted to:

```
export type PartialAlert = Pick<Alert, 'id'> & Partial<Alert>;
```

Seems like TS usually does a good job allowing some "slop" here - basically, `id` is defined twice - required on the left type, optional on the right, so . ... it's required in the final type, and the types of `id` are the same so ... "it works".  Not quite sure tho :-)

I feel like there must be a good, simple pattern for building and naming (heh) these sort of structures, but I've not quite figured it out yet ...

Reply:
Sadly no, because that results in `id` becoming `string | undefined` which is exactly what we don't want here...